src/server/dflycmd.cc

```diff
@@ -502,6 +504,9 @@ OpStatus DflyCmd::StartFullSyncInThread(FlowInfo* flow, Context* cntx, EngineSha
   DCHECK(!flow->full_sync_fb.IsJoinable());
   DCHECK(shard);
 
+  if (flow->conn == nullptr)
```

Comment:
This PR partially addresses the issue above by checking `conn` in critical places. Not sure it's enough.

Reply:
Again, we hold the mutex along calling StartFullSyncInThread

Reply:
Redundant defensive checks can make the code more confusing because they let the reader question his understanding of the state machine. Currently it should not be possible as far as I remember (haven't edited this code for ages)

Reply:
ok, I will replace them with DCHECKs